src/skeleplex/graph/constants.py

```diff
@@ -10,4 +10,13 @@
 END_NODE_KEY = "end_node"
 GENERATION_KEY = "level"
 VALIDATION_KEY = "validated"
-LENGTH_KEY = "length"
\ No newline at end of file
+LENGTH_KEY = "length"
+NUMBER_OF_TIPS_KEY = "number_of_tips"
+SISTER_EDGE_KEY = "sister_edge"
+PARENT_EDGE_KEY = "parent_edge"
+DAUGHTER_EDGES_KEY = "daughter_edges"
+ROTATION_ANGLE_EDGE_KEY = "rotation_angle"
+BRANCH_ANGLE_EDGE_KEY = "branch_angle"
+SIBLING_ANGLE_EDGE_KEY = "sibling_angle"
```

Comment:
Long term, it might be nice to replace these edge attributes with operations on the graph (e.g., when you need the daughter, just compute it from the graph). I can see how this would speed things up by caching results. However, it also adds overhead because we need to keep these properties in sync with the graph (I.e., the daugher edge property could be set to be different than the daughters as defined by the graph). Basically, it is best if there is one source of ground truth. Storing duplicate information in different places is generally something to be cautious about because it is easy to create invalid/contradicting states. 

Of course, this would require us using directed graphs. Not something we need to solve in this PR, but something to think about as we move forward.

Reply:
Ok, I keep it for now, but will start computing it directly from the graph and not to rely on the attributes. In the past, I really liked storing measurments on the graph, because its easy to come back to and to visualize things, but I can see the risk.